lib/brook/driver/kafka/serializer.ex

```diff
@@ -36,16 +36,30 @@ defimpl Brook.Deserializer.Protocol, for: Brook.Event do
 
   def deserialize(%Brook.Event{}, data) do
     data
+    |> get_struct()
     |> deserialize_data()
     |> to_struct()
   end
 
-  defp deserialize_data(decoded_json) do
-    case Brook.Deserializer.deserialize(decoded_json.data) do
+  defp deserialize_data({:ok, data_struct, decoded_json}) do
+    case Brook.Deserializer.deserialize(data_struct, decoded_json.data) do
       {:ok, value} -> {:ok, Map.put(decoded_json, :data, value)}
       error_result -> error_result
     end
   end
+  defp deserialize_data({:error, _reason} = error), do: error
+
+  defp get_struct(%{__struct__: custom_struct} = data) do
+    struct_module = String.to_atom(custom_struct)
+    Code.ensure_loaded(struct_module)
```

Comment:
So if this module doesn't exist (and can't be loaded), we just error out? Seems legit, just curious.

Reply:
Correct